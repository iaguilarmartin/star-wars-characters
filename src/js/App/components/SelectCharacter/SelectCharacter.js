/* eslint-disable jsx-a11y/label-has-associated-control, jsx-a11y/label-has-for */

import React from 'react';
import PropTypes from 'prop-types';

import { Section, Select } from './SelectCharacter.styles';
import charactersData from './characters.json';

const options = charactersData.characters.map(({ name, url }) => ({
  value: url,
  label: name
}));

const SelectCharacter = ({ onSelectCharacter }) => (
  <Section>
    <label htmlFor="charactersSelect">Select a character:</label>
    <Select
      classNamePrefix="react-select"
      onChange={onSelectCharacter}
      inputId="charactersSelect"
      placeholder="List of characters"
      options={options}
    />
  </Section>
);

SelectCharacter.propTypes = {
  onSelectCharacter: PropTypes.func.isRequired
};

export default SelectCharacter;
