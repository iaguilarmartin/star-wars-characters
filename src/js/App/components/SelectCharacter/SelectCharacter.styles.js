import ReactSelect from 'react-select';
import styled from '@emotion/styled';
import { css } from '@emotion/core';

import { space } from '../../styles/spacing';
import fonts from '../../styles/fonts';
import colors from '../../styles/colors';
import { mediaQueries } from '../../styles/breakpoints';

export const Section = styled.section`
  display: flex;
  flex-direction: column;
  margin: 0 ${space.x1};

  ${mediaQueries.md(css`
    align-items: center;
    flex-direction: row;
  `)}
`;

export const Select = styled(ReactSelect)`
  margin-top: ${space.x1};
  font-family: ${fonts.Starjedi};
  outline: none;

  ${mediaQueries.md(css`
    width: 350px;
    margin-top: 0;
    margin-left: ${space.x1};
  `)}

  .react-select__control {
    background-color: ${colors.greyDark};
    border-color: ${colors.grey};
  }

  .react-select__value-container {
    outline: none;
  }

  .react-select__control--is-focused {
    border-color: ${colors.yellow} !important;
    box-shadow: 0 0 0 1px ${colors.yellow};
  }

  .react-select__single-value {
    color: ${colors.white};
  }

  .react-select__placeholder {
    font-size: ${fonts.sizes.s};
    font-style: italic;
  }

  .react-select__menu {
    background-color: ${colors.greyDark};
  }

  .react-select__option--is-focused {
    background-color: ${colors.grey};
  }

  .react-select__option--is-selected {
    background-color: ${colors.blue};
  }
`;
