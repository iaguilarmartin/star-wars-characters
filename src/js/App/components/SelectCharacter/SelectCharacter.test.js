import React from 'react';
import { shallow } from 'enzyme';
import toJSON from 'enzyme-to-json';

import SelectCharacter from './SelectCharacter';

describe('<LoadingCharacter/>', () => {
  it('renders', () => {
    shallow(<SelectCharacter onSelectCharacter={() => {}} />);
  });

  it('matches the snapshot', () => {
    const wrapper = shallow(<SelectCharacter onSelectCharacter={() => {}} />);
    expect(toJSON(wrapper)).toMatchSnapshot();
  });

  it('calls handler function on select character', () => {
    const mockHandler = jest.fn();
    const wrapper = shallow(
      <SelectCharacter onSelectCharacter={mockHandler} />
    );
    wrapper.find('Select').simulate('change');
    expect(mockHandler).toHaveBeenCalled();
  });
});
