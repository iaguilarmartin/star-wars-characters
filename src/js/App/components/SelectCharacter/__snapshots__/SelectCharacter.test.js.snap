// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<LoadingCharacter/> matches the snapshot 1`] = `
<Section>
  <label
    htmlFor="charactersSelect"
  >
    Select a character:
  </label>
  <Select
    classNamePrefix="react-select"
    defaultInputValue=""
    defaultMenuIsOpen={false}
    defaultValue={null}
    inputId="charactersSelect"
    onChange={[Function]}
    options={
      Array [
        Object {
          "label": "Luke Skywalker",
          "value": "https://swapi.co/api/people/1/",
        },
        Object {
          "label": "Darth Vader",
          "value": "https://swapi.co/api/people/4/",
        },
        Object {
          "label": "Obi-wan Kenobi",
          "value": "https://swapi.co/api/people/unknown/",
        },
        Object {
          "label": "R2-D2",
          "value": "https://swapi.co/api/people/3/",
        },
      ]
    }
    placeholder="List of characters"
  />
</Section>
`;
